src/test/java/de/tum/in/www1/artemis/service/AutomaticTextFeedbackServiceTest.java

```diff
@@ -0,0 +1,440 @@
+package de.tum.in.www1.artemis.service;
+
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import de.tum.in.www1.artemis.AbstractSpringIntegrationBambooBitbucketJiraTest;
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.enumeration.FeedbackType;
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+import de.tum.in.www1.artemis.exception.NetworkingError;
+import de.tum.in.www1.artemis.repository.*;
+import de.tum.in.www1.artemis.service.connectors.TextFeedbackValidationService;
+import de.tum.in.www1.artemis.util.TextExerciseUtilService;
+
+public class AutomaticTextFeedbackServiceTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    // SUT
+    private AutomaticTextFeedbackService automaticTextFeedbackService;
+
+    @Autowired
+    private TextClusterRepository textClusterRepository;
+
+    @Autowired
+    private TextExerciseRepository textExerciseRepository;
+
+    @Autowired
+    private FeedbackRepository feedbackRepository;
+
+    @Autowired
+    private TextExerciseUtilService textExerciseUtilService;
+
+    @Autowired
+    private ResultRepository resultRepository;
+
+    @Autowired
+    private TextSubmissionRepository textSubmissionRepository;
+
+    @Autowired
+    private StudentParticipationRepository studentParticipationRepository;
+
+    @Autowired
+    private TextClusterUtilityService textClusterUtilityService;
+
+    @Mock
+    private TextFeedbackValidationService mockedTextFeedbackValidationService;
```

Comment:
please be careful here, we have a policy on MockBeans and SpyBeans.
Not sure if "@Mock" also enforces the server to restart (twice) during the tests which significantly increases the test execution time. Please double check this